xbmc/windowing/X11/WinSystemX11GLContext.cpp

```diff
@@ -247,56 +247,47 @@ bool CWinSystemX11GLContext::RefreshGLContext(bool force)
 
   CScreenshotSurfaceGL::Register();
 
-  std::string gpuvendor;
-  const char* vend = (const char*) glGetString(GL_VENDOR);
-  if (vend)
-    gpuvendor = vend;
-  std::transform(gpuvendor.begin(), gpuvendor.end(), gpuvendor.begin(), ::tolower);
-  bool isNvidia = (gpuvendor.compare(0, 6, "nvidia") == 0);
-  bool isIntel = (gpuvendor.compare(0, 5, "intel") == 0);
-  std::string gli = (getenv("KODI_GL_INTERFACE") != nullptr) ? getenv("KODI_GL_INTERFACE") : "";
-
-  if (gli != "GLX")
+  //! @todo: move to using a runtime switch param instead of an env variable?
+  std::string gli = (getenv("KODI_GL_INTERFACE") != nullptr) ? getenv("KODI_GL_INTERFACE") : "EGL";
```

Comment:
The todo becomes done on following requests where the select about becomes added?

Reply:
No I haven't done this yet. The problem here is that currently we only use EGL for intel unless EGL is explicitly selected via the env variable (this is only on X11). This behaviour allows amd to use vdpau by default unless selecting EGL (and thus using VAAPI).

The inherent problem that removing the `glGetString` call creates is that this behaviour won't occur for AMD (as it will always try EGL first). So we can do two things in my opinion:
1) add a runtime switch like `--interop=egl / --interop=glx` or `--egl` or `--glx`. This would allow someone to select which context they want to use (You can do this already with the `KODI_GL_INTERFACE` env variable but meh.
2) find another way to query the graphics card in use (through x11) to allow the current behaviour to continue.